trino-aws-proxy/src/test/java/io/trino/aws/proxy/server/rest/TestAwsChunkedInputStream.java

```diff
@@ -0,0 +1,337 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.aws.proxy.server.rest;
+
+import com.google.common.io.ByteStreams;
+import io.trino.aws.proxy.server.signing.TestingChunkSigningSession;
+import io.trino.aws.proxy.spi.credentials.Credential;
+import io.trino.aws.proxy.spi.signing.ChunkSigningSession;
+import jakarta.ws.rs.WebApplicationException;
+import org.junit.jupiter.api.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class TestAwsChunkedInputStream
+{
+    private static final Credential GOOD_CREDENTIAL = new Credential("TEST_ACCESS_KEY", "TEST_SECRET_KEY");
+    private static final String GOOD_CONTENT = "The quick brown fox jumps over the lazy dog's head";
+    private static final String GOOD_SEED = "10e300fbbad6946a482aaa7de0bdc8f592d4c372306dd746a22d18b7b66b4527";
```

Comment:
Nit: this seed can be anything, as long as it isn't the same as `BAD_SEED`. Unlike in other tests, this one isn't validated at all (the seed is the signature of the request itself, which we don't check here).

What about we do `GOOD_SEED = "GOOD_DUMMY_VALUE"` and `BAD_SEED = "BAD_DUMMY_VALUE"`?

I think it makes it easier to understand what exactly `AwsChunkedInputStream` will validate and what it will not validate. GIving it a seemingly "real" seed could be a bit confusing IMO